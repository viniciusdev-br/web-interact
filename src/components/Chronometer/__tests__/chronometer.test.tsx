import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { describe, expect, it } from "vitest";
import { Chronometer } from '..';

function toJson(component: renderer.ReactTestRenderer) {
    const result = component.toJSON()
    expect(result).toBeDefined()
    expect(result).not.toBeInstanceOf(Array)
    return result as renderer.ReactTestRendererJSON
}

describe('Chronometer', () => {
    it('Renderiza componente de Chronometro', () => {
        render(<Chronometer />);
        screen.debug();
        const component = renderer.create(<Chronometer />);
        const tree = toJson(component);
        expect(tree).toMatchSnapshot();
    });
});