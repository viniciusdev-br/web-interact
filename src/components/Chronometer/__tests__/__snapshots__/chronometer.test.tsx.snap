// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Chronometer > Renderiza componente de Chronometro 1`] = `
<span
  className="time-service"
>
  00:00
</span>
`;

exports[`Chronometer > Renderiza componente de Chronometro iniciando com tempo 00:00 1`] = `
<span
  className="time-service"
>
  00:00
</span>
`;
